internal/components/datasource/controller.go

```diff
@@ -5,13 +5,12 @@ import (
 	"errors"
 	"time"
 
+	"github.com/grafana/grafana/pkg/models"
```

Comment:
I think it might be better if we avoid creating dependencies from new models to old models, on the premise that the easy migration path in the future will probably entail creating dependencies from old models to new models.

Not sure, though.

Reply:
I agree, I think this is temporary because the `Store.Get()` implementation calls the old `SQLStore.GetDataSource()` which returns `models.ErrDataSourceNotFound`.
I expect that we will change this at some point following [this](https://github.com/grafana/grafana/pull/44771#discussion_r801699457) discussion.